tests/unit/s2n_x509_validator_test.c

```diff
@@ -993,9 +989,9 @@ int main(int argc, char **argv) {
         EXPECT_SUCCESS(read_file(&ocsp_data_stuffer, S2N_OCSP_RESPONSE_DER, S2N_MAX_TEST_PEM_SIZE));
         uint32_t ocsp_data_len = s2n_stuffer_data_available(&ocsp_data_stuffer);
         EXPECT_TRUE(ocsp_data_len > 0);
-        EXPECT_EQUAL(S2N_CERT_ERR_EXPIRED, s2n_x509_validator_validate_cert_stapled_ocsp_response(&validator, connection,
-                                                                                          s2n_stuffer_raw_read(&ocsp_data_stuffer, ocsp_data_len),
-                                                                                          ocsp_data_len));
+        EXPECT_ERROR_WITH_ERRNO(s2n_x509_validator_validate_cert_stapled_ocsp_response(&validator, connection,
+                s2n_stuffer_raw_read(&ocsp_data_stuffer, ocsp_data_len), ocsp_data_len), S2N_ERR_CERT_UNTRUSTED);
```

Comment:
I'm a little worried we're losing information here. Should we add a more specific S2N_ERR_CERT_EXPIRED error code? Same with other tests that previously checked for non-OK error codes.

Reply:
The fact that we had no way to actually communicate the error codes up to the application and they were all lumped under "cert untrusted", despite us knowing what was actually wrong, was one of the annoying aspects of this code.

Reply:
I added a bunch of new error codes in dd6b143db08d592ed8485c793845495093d4bbbe. I went through both functions and changed the error codes to what I thought would make more sense.